### PR TITLE
Complete post/slides endpoint

### PR DESCRIPTION
**Pull request by fadezak100**
- Created a new route for slides
- Created a new controller related to slides
- Test slide controller

### CODE REVIEW

File: src/controllers/SlideController.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Response } from 'express'
+import { CreatePresentationRequest } from '../interfaces/SlideInterface'
+import SlideService from '../services/SlideService'
+
+const createPresentation = async (
+  req: CreatePresentationRequest,
+  res: Response
+) => {
+  try {
+    const { link, title, isPrivate, isLive } = req.body
```
Comment (AhmedSafi97) [inline src/controllers/SlideController.ts:10]: where is validation ya man

File: src/controllers/SlideController.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      title,
+      isPrivate,
+      isLive
+    })
+
+    res.status(200).send({ message: 'success', presentation })
+  } catch (error) {
+    res.status(400).send({ error: 'bad request' })
+  }
+}
```
Comment (AhmedSafi97) [inline src/controllers/SlideController.ts:22]: check how @Zayan-Alaraishy is handling the errors here https://github.com/CA-G12/mern-presently.io/pull/10

File: src/interfaces/SlideInterface.ts [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+
+export interface CreatePresentationOptions {
+  title: string
+  link: string
+  isLive?: boolean
+  isPrivate?: boolean
+}
+
+export interface CreatePresentationRequest extends Request {
+  body: { title: string; link: string; isPrivate?: boolean; isLive?: boolean }
```
Comment (AhmedSafi97) [inline src/interfaces/SlideInterface.ts:19]: DRY your code by reusing some interfaces `body: CreatePresentationOptions`

File: src/repositories/SlideRepository.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import Slide from '../models/SlideModel'
+import { CreatePresentationOptions } from '../interfaces/SlideInterface'
+
+const createPresentation = (options: CreatePresentationOptions) =>
```
Comment (AhmedSafi97) [inline src/repositories/SlideRepository.ts:4]: this should be called `presentation`
```js
Slide.create(presentation)
```

**Review by AhmedSafi97 — CHANGES_REQUESTED** [review]
Looks good, small changes then we are good to go

**Review by moustf — COMMENTED** [review]
To be honest, I can not request any changes because I do not understand your code style, but thanks for putting me as a reviewer

**Comment by fadezak100** [comment]
> To be honest, I can not request any changes because I do not understand your code style, but thanks for putting me as a reviewer

Thank you for putting the time to review the code Mustafa! I appreciate it.

**Review by AhmedSafi97 — DISMISSED** [review]

**Review by nadasuhailAyesh12 — APPROVED** [review]

**Review by Zayan-Alaraishy — DISMISSED** [review]

**Pull request merged**